### PR TITLE
Extend CRUD API

### PR DESCRIPTION
**Pull request by gektin**
Still in the process of reworking tests from legacy codebase, but have tested these endpoints in Postman.

### README.md
The changes here are fairly simple, just documenting the API that has been built out below.

### client/relatedData.jsx
More simple changes. I've added new API calls here, so as not to have to deal w/ recursive server routing for the time being.

### relatedData.js
I've built out the database to have methods for all the CRUD operations. I've also made the data more complex based on some discussions with Leslie. 

### seedData.js
I've updated the seeding script to work with my newly designed data.

### server/index.js
I've added CrUD methods and rewritten the cRud method in order to work with my data.

### CODE REVIEW

File: seedData.js [hunk 1]
```diff
@@ -1,52 +1,36 @@
...
-                })
-              }
-            }
-          }
-        })
-      }
-    })
-  }
+  return new Promise((resolve, reject) => {
+    drop();
```
Comment (TolyK) [inline seedData.js:10]: I think this drop method is a mongo specific command

File: seedData.js [hunk 2]
```diff
@@ -1,52 +1,36 @@
...
+        reposts: Math.ceil(Math.random() * 100),
+        comments: Math.ceil(Math.random() * 100),
+        genre: genners[Math.floor(Math.random() * genners.length)],
+        onPlaylists: [Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100)],
+        producedBy: producers[Math.floor(Math.random() * producers.length)],
+        likedBy: [Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100)],
+        playedBy: [Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100)],
+        repostedBy: [Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100)],
+      };
+      addTrack(current, () => {});
```
Comment (TolyK) [inline seedData.js:28]: for the big seed file & import w/ 10m, you probably want to bulk import, not use addTrack each time

File: server/index.js [hunk 3]
```diff
@@ -2,12 +2,17 @@ const express = require('express');
 const path = require('path');
 const chalk = require('chalk');
 const axios = require('axios');
```
Comment (TolyK) [inline server/index.js]: not used here right?

File: server/index.js [hunk 4]
```diff
@@ -17,60 +22,94 @@ app.use('/', expressStaticGzip(path.join(__dirname, '../public'), {
...
+
+app.delete('/relatedTracks/:song', (req, res) => {
+  let song = req.params.song;
+  deleteTrack(song, (err, track) => {
+    if (err) {
+      console.log(`delete error: ${err}`);
+      res.status(500).end('we had trouble deleting that track; try again later');
+    } else {
+      if (track === null) {
+        res.status(404).send('no such track');
```
Comment (TolyK) [inline server/index.js:105]: really great use of 404 code, I need to do this

**Review by TolyK — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -1 +1,91 @@
...
+  plays: Number,
+  likes: Number,
+  reposts: Number,
+  comments: Number,
+  genre: String,
+  onPlaylists: [Number],
+  producedBy: String,
+  likedBy: [Number],
+  playedBy: [Number],
+  repostedBy: [Number],
```
Comment (redperezz) [inline README.md:15]: I like that you laid out the shape of the data, very useful.👍🏽

File: README.md [hunk 6]
```diff
@@ -1 +1,91 @@
...
+        "song_id": [id],
+        "plays": 5258,
+        "likes": 766,
+        "reposts": 83,
+        "onPlaylists": [67,71,10],
+        "likedBy": [65,89,4],
+        "playedBy": [83,20,41],
+        "repostedBy": [2,84,81],
+        "genre": "spoken",
+        "producedBy": "zaytoven",
```
Comment (redperezz) [inline README.md:35]: Nice example!

File: README.md [hunk 7]
```diff
@@ -1 +1,91 @@
...
+        "onPlaylists": [67,71,10],
+        "likedBy": [65,89,4],
+        "playedBy": [83,20,41],
+        "repostedBy": [2,84,81],
+        "genre": "ambient",
+        "producedBy": "stock boy",
+      }
+
+Response body:
+      "track [id] posted"
```
Comment (redperezz) [inline README.md:57]: Great response message, reminded me to include this on top of a status code.👌🏽

File: README.md [hunk 8]
```diff
@@ -1 +1,91 @@
...
+
+Example: http://localhost:3001/relatedTracks/[id]
+
+Request body:
+      {
+        "song_id": [id],
+      }
+
+Response body:
+      "track [id] deleted"
```
Comment (redperezz) [inline README.md:91]: ✓ well done readme!

File: client/relatedData.jsx [hunk 9]
```diff
@@ -15,11 +16,30 @@ class RelatedTracks extends React.Component {
...
+      if (index === 0) {
+        this.setState({
+          related: [data],
+        });
+      } else {
+        this.setState({
+          related: this.state.related.concat(data),
+        });
+      }
+    })
```
Comment (redperezz) [inline client/relatedData.jsx:39]: ✓ very nice!

File: relatedData.js [hunk 10]
```diff
@@ -1,7 +1,10 @@
 const mongoose = require('mongoose');
 const chalk = require('chalk');
 
-mongoose.connect('mongodb://localhost/test');
+mongoose.connect('mongodb://localhost/tracks', {
+                                                  useUnifiedTopology: true,
+                                                  useNewUrlParser: true,
+                                                });
```
Comment (redperezz) [inline relatedData.js:7]: ✓

File: relatedData.js [hunk 11]
```diff
@@ -12,29 +15,44 @@ db.once('open', function() {
...
   likes: Number,
   reposts: Number,
-  related: [Number]
+  comments: Number,
+  genre: String,
+  onPlaylists: [Number],
+  producedBy: String,
+  likedBy: [Number],
+  playedBy: [Number],
+  repostedBy: [Number],
```
Comment (redperezz) [inline relatedData.js:28]: Oh wow! Great job making the data more complex!👌🏽

File: relatedData.js [hunk 12]
```diff
@@ -12,29 +15,44 @@ db.once('open', function() {
...
+      let track = new Track(trackData);
+      track.save((err, track) => {
+        if (err) {
+          console.log(chalk.red('Problem saving track', err));
+          cb(err);
+        } else {
+          console.log(chalk.blue(`Track ${track.song_id} saved`));
+          cb(null, track);
+        }
+      })
```
Comment (redperezz) [inline relatedData.js:48]: ✓ good changes.

File: relatedData.js [hunk 13]
```diff
@@ -44,7 +62,18 @@ let findTrack = function(id, cb) {
...
+    if (err) {
+      cb(err);
+    } else {
+      console.log(track)
+      cb(null, track);
+    }
+  });
+};
+
+const deleteTrack = function (id, cb = () => {}) {
```
Comment (redperezz) [inline relatedData.js:76]: ✓

File: relatedData.js [hunk 14]
```diff
@@ -56,6 +85,27 @@ let deleteTrack = function (id, cb) {
...
+
+
+module.exports = {
+  addTrack,
+  findTrack,
+  updateTrack,
+  deleteTrack,
+  drop,
+  close,
+}
```
Comment (redperezz) [inline relatedData.js:111]: ✓ Well done! Looking forward to seeing where the drop and close functions are being called.

**Review by redperezz — APPROVED** [review]
It looks like Toly also reviewed this PR lol, either way great job!

File: seedData.js (hunk 2 above)
Comment (gektin) [reply seedData.js:28]: Yep, I'm doing something similar to you. Writing csv files and just importing those directly.

File: seedData.js (hunk 1 above)
Comment (gektin) [reply seedData.js:10]: It is, so it'll go away eventually.

**Pull request merged**